### PR TITLE
Wired up Site Permissions Panel to API

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Wired up Site Permissions Panel to `BraveWalletService` api's

1) Build out the `Site Permissions` panel and needed components.
2) Added state `connectedAccounts` in the wallet reducer that will return all accounts that have permissions for the `activeOrigin`
3) Added a check to the `Main` wallet panel to display `Connected` if selected account is included in the `connectedAccounts` list.
4) Added a `WalletAction` for removing a site permission for an account.
5) Connect with site panel will now have the `selectedAccount` auto selected.
6) Localized Wallet Panel Titles

Resolves <https://github.com/brave/brave-browser/issues/18453>
Resolves <https://github.com/brave/brave-browser/issues/18372>
Resolves <https://github.com/brave/brave-browser/issues/18657>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/136492181-52d4eb21-740f-4e1a-bf3e-7e3c62a3e811.mov

### CODE REVIEW

File: components/brave_wallet_ui/options/panel-titles.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { PanelTitleObjectType } from '../constants/types'
+import { getLocale } from '../../common/locale'
+
+export const PanelTitles = (): PanelTitleObjectType[] => [
```
Comment (onyb) [inline components/brave_wallet_ui/options/panel-titles.ts:4]: Assuming we really need this as a function for lazy evaluation, let's memoize wherever this is called in components.

File: components/brave_wallet_ui/components/extension/connected-panel/index.tsx [hunk 2]
```diff
@@ -99,7 +103,7 @@ const ConnectedPanel = (props: Props) => {
       />
       <CenterColumn>
         <StatusRow>
-          <OvalButton onClick={connectAction}>
+          <OvalButton disabled={activeOrigin === 'chrome://wallet/'} onClick={onShowSitePermissions}>
```
Comment (onyb) [inline components/brave_wallet_ui/components/extension/connected-panel/index.tsx:106]: Is there a possibility that the origin inside the wallet changes in future? Wondering if `.startsWith()` makes sense in that case.

File: components/brave_wallet_ui/panel/container.tsx [hunk 3]
```diff
@@ -381,6 +383,18 @@ function Container (props: Props) {
     // Logic here to cancel connecting your hardware wallet
   }
 
+  const removeSitePermission = (origin: string, address: string) => {
+    props.walletActions.removeSitePermission({ origin: origin, account: address })
+  }
+
+  const isConnectedToSite = React.useMemo((): boolean => {
+    if (activeOrigin === 'chrome://wallet/') {
```
Comment (onyb) [inline components/brave_wallet_ui/panel/container.tsx:391]: `'chrome://wallet/'` was used somewhere else too, so can we move this to a common place and give it a meaningful name?

File: components/brave_wallet_ui/panel/container.tsx [hunk 4]
```diff
@@ -381,6 +383,18 @@ function Container (props: Props) {
...
+      return true
+    } else {
+      return connectedAccounts.includes(selectedAccount)
```
Comment (onyb) [inline components/brave_wallet_ui/panel/container.tsx:394]: Probably a gap in my JS knowledge, but is this operation safe? I don't know why it works here, but here's a small snippet to demonstrate my point.

```js
> [{foo: 'bar'}].includes({foo: 'bar'})
false
```

File: components/brave_wallet_ui/components/extension/connected-panel/index.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/connected-panel/index.tsx]: `activeOrigin` will always return the origin for an active tabs `url` so for instance
```
chrome://wallet/crypto/portfolio would return chrome://wallet/
```
So `startsWith()` would not make a difference in this case!

File: components/brave_wallet_ui/common/async/wallet_async_handler.ts [hunk 5]
```diff
@@ -207,14 +208,27 @@ async function refreshWalletInfo (store: Store) {
     }
   }))
   store.dispatch(WalletActions.setTransactionList(getTransactions))
+
+  // Get a list of accounts with permissions of the active origin
+  const accounts = state.accounts
+  const origin = state.activeOrigin
+  const getAllPermissions = await Promise.all(accounts.map(async (account) => {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/async/wallet_async_handler.ts:216]: Not in this instance, we still need to map over each account and get the permission first and then return the account if permission is true.
Doing only a filter would always return all accounts.

File: components/brave_wallet_ui/options/panel-titles.ts (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/options/panel-titles.ts:4]: As discussed via slack, for a timing issue related to `getLocale` this has to be a function. Not able to memoize this in the reducer where it's being called.

File: components/brave_wallet_ui/panel/container.tsx (hunk 4 above)
Comment (onyb) [reply components/brave_wallet_ui/panel/container.tsx]: Nice use of `.some`!

File: components/brave_wallet_ui/common/async/wallet_async_handler.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (onyb) [reply components/brave_wallet_ui/common/async/wallet_async_handler.ts:216]: I was thinking something like this. Does this work?

```ts
const accountsWithPermission = await Promise.all(accounts.filter(async (account) => {
  const result = await braveWalletService.hasEthereumPermission(origin, account.address)
  return result.hasPermission
}))
```

**Review by onyb — APPROVED** [review]
LGTM

File: components/brave_wallet_ui/common/async/wallet_async_handler.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/async/wallet_async_handler.ts:216]: Tried that and it always returns all accounts.. :/

**Pull request merged**